src/sagemaker_sklearn_extension/feature_extraction/sequences.py

```diff
@@ -0,0 +1,414 @@
+    and then TSFreshFeatureExtractor extracts time series features from each list. The outputs from each column are then
+    stacked horizontally into a single array.
+
+    See TSFlattener and TSFreshFeatureExtractor for more details.
+
+    Parameters
+    ----------
+    max_allowed_length : int (default = 10000)
+        Maximum allowed length of an input sequence. If the length of a sequence is greater than ``max_allowed_length``,
+        the transformer will truncate its beginning or end as indicated by ``trim_beginning``.
+
+    trim_beginning : bool (default = True)
+        If a sequence length exceeds ``max_allowed_length``, trim its start and only keep the last `max_allowed_length``
+        values if ``trim_beginning`` = True, otherwise trim the end and only keep the first max_allowed_length values.
+
+    augment : boolean (default=True):
+        Whether to append the tsfresh features to the original data (if True),
+        or output only the extracted tsfresh features (if False).
+        If True, also pad shorter sequences (if any) in the original data with np.nans, so that all sequences
+        match the length of the longest sequence, and interpolate them as indicated by ``interpolation_method``.
+
+    interpolation_method : {'linear', 'fill', 'zeroes', 'hybrid', None} (default='hybrid')
+        'linear': linear interpolation
+        'fill': forward fill to complete the sequences; then, backfill in case of NaNs at the start of the sequence.
+        'zeroes': pad with zeroes
+        'hybrid': replace with zeroes any NaNs at the end or start of the sequences, and forward fill NaNs in between
+         None: no interpolation
+
+    extraction_type : {'minimal', 'efficient', 'all'} (default='efficient')
+        Control the number of features extracted from tsfresh.
+        'minimal': most of the feature extractors are disabled and only a small subset is used
+        'efficient': extract 781 tsfresh features, namely all of them except for the expensive to compute ones
+        'all': extract all 787 tsfresh features
+
+    Examples
+    --------
+    >>> from sagemaker_sklearn_extension.feature_extraction.sequences import TimeSeriesFeatureExtractor
+    >>> import numpy as np
+    >>> data = [["1,, 3, 44", "3, 4, 5, 6"], ["11, 111", "1, 1, 2, 2"], ["NaN, , 1, NaN", "2, 1, 3, 2"]]
+    >>> ts_pipeline = TimeSeriesFeatureExtractor(augment=True)
+    >>> X = ts_pipeline.fit_transform(data)
+    >>> print(X.shape)
+    (3, 1570)
+    >>> ts_pipeline = TimeSeriesFeatureExtractor(augment=False)
+    >>> X = ts_pipeline.fit_transform(data)
+    >>> print(X.shape)
+    (3, 1562)
+    """
+
+    def __init__(
+        self,
+        max_allowed_length=10000,
+        trim_beginning=True,
+        augment=True,
+        interpolation_method="hybrid",
+        extraction_type="efficient",
+    ):
+        super().__init__()
+        assert max_allowed_length > 0, f"{max_allowed_length} must be positive."
```

Comment:
This should be a ValueError, see other estimators

Reply:
Thanks for the callout. Done.